Source/MQTTnet/Internal/Memory.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace MQTTnet.Internal
+{
+    public static class Memory
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void Copy(byte[] source, int sourceIndex, byte[] destination, int destinationIndex, int length)
```

Comment:
```suggestion
        public static void Copy(ReadOnlySpan<byte> source, Span<byte> destination)
```
? Makes this API simpler and more future-proof.

For ease of use at callsites this could be another overload where the byte[]-based one just forwards to the span-based one.